receiver/nginxreceiver/internal/record/recorder.go

```diff
@@ -0,0 +1,55 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package record
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/nginxreceiver/internal/metadata"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/nginxreceiver/internal/model"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+// Item extracts data from NGINX Access Items and records them using the given MetricsBuilder.
+func Item(ai *model.NginxAccessItem, mb *metadata.MetricsBuilder) error {
```

Comment:
Open to ideas for a better name for this, currently this is being called as `record.Item(...)`, which does an okay (but not great) job of conveying what it's doing.